zipkin-api.yaml

```diff
@@ -0,0 +1,259 @@
+          lookback is 7 days, traces returned should be those nearest to 10:20
+          today, not 10:20 a week ago.
+
+          Time units of endTs and lookback are milliseconds as opposed to
+          microseconds, the grain of Span.timestamp. Milliseconds is a more
+          familiar and supported granularity for query, index and windowing
+          functions
+      parameters:
+        - name: serviceName
+          in: query
+          required: true
+          description: service name
+          type: string
+        - name: annotationQuery
+          in: query
+          type: string
+          required: true
+          description: |
+             Ex. `http.uri=/foo and retried` - If key/value (has an `=`),
+             constrains against Span.binaryAnnotations of time string. If just
+             a word, constrains against Span.annotations. Any values are AND
+             against eachother. This means a span in the trace must match all of
+             these.
+        - name: minDuration
+          in: query
+          type: integer
+          description: |
+             Ex. 100000 (for 100ms). Only return traces whose `Span.duration` is
+             greater than or equal to minDuration microseconds.
+        - name: maxDuration
+          in: query
+          type: integer
+          description: |
+            Only return traces whose Span.duration is less than or equal to
+            `maxDuration` microseconds. Only valid with minDuration.
+        - name: endTs
+          in: query
+          type: string
+          description: |
+            Only return traces where all Span.timestamp are at or before this
+            time in epoch milliseconds. Defaults to current time.
+        - name: lookback
+          type: string
+          in: query
+          description: |
+            Only return traces where all  Span.timestamp are at or after (endTs
+            - * lookback) in milliseconds. Defaults to endTs, limited to a
+            system parameter QUERY_LOOKBACK
+
+        - name: limit
+          in: query
+          default: 10
+          type: integer
+          description: |
+            Maximum number of traces to return. Defaults to 10
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: "#/definitions/ListOfTracesLists"
+  /trace/{traceId}:
+    parameters:
+      - name: traceId
+        in: path
+        required: true
+        description: the 64-bit hex-encoded id of the trace as a path parameter.
+        type: string
+    get:
+      responses:
+        200:
+          description: OK
+          schema:
+            description: An array of spans (annotations?)
+            type: array
+            items:
+              $ref: "#/definitions/Annotation"
+        404:
+           description: "`traceId` not found"
+  /spans:
+    post:
+      description: Uploads a list of spans in json format
+      consumes:
```

Comment:
Not sure about this. Till now I was thinking even empty body response considered plain text or if it's a JSON API an `application/json`. We can put `produces: []` to say this endpoint produces nothing though. I'm not sure how the rest of Swagger tooling thinks about it.